fem/linearform.hpp

```diff
@@ -101,6 +101,19 @@ class LinearForm : public Vector
        and GridFunction. */
    double operator()(const GridFunction &gf) const { return (*this)*gf; }
 
+   /// Redefine '=' for LinearForm = constant.
+   LinearForm &operator=(double value);
+
+   /// Copy the data from @a v.
+   /** The size of @a v must be equal to the size of the FiniteElementSpace
+       @a fes. */
+   LinearForm &operator=(const Vector &v);
+
+   /// Copy the data from @a v.
+   /** The LinearForms @a v and @a *this must have FiniteElementSpaces with
+       the same size. */
+   LinearForm &operator=(const LinearForm &v);
+
```

Comment:
I think we do not need to overload `operator=` for `const LinearForm &` because the version for `const Vector &` will be called when the rhs is a `LinearForm` and we do not need different behavior when the rhs is a `LinearForm`.

I know that we have this overload in the classes `(Par)GridFunction` (for rhs of type `(Par)GridFunction`) but I think we can safely remove those.

Reply:
I agree.  I'll remove all four methods for consistency.

Reply:
@mlstowell,

I think what I said above is incorrect: if we do not define the copy assignment operator
```c++
   LinearForm &LinearForm::operator=(const LinearForm &);
```
then the compiler defines one implicitly, see [copy assignment operator](https://en.cppreference.com/w/cpp/language/copy_assignment). Therefore, if one uses
```c++
   linear_form_1 = linear_form_2;
```
then the implicitly defined copy assignment will be called and NOT the method
```c++
   LinearForm &LinearForm::operator=(const Vector &v);
```
This also applies to the other classes: `(Par)GridFunction`, `(Par)LinearForm`.

This was my mistake - I'm looking at #568 to address #655 and I just realized I made a mistake here. I'll add the missing copy assignment operators to that PR (#568).